server/exposures/cmaq.py

```diff
@@ -35,37 +34,14 @@ def is_valid_date_range(self, **kwargs):
             return True
 
     def is_valid_lat_lon(self, **kwargs):
-        (valid_points, message, pt) = self.validate_coordinate_point(**kwargs)
-	
-        return valid_points
-        
-    def validate_coordinate_point(self, **args):
-        session = Session()
-        latlon = session.query(CmaqExposuresList.resolution).filter(
-            CmaqExposuresList.latLon == args.get('exposureType')).one()
-        session.close()
-
-        # if lat is populated - make sure lon is too - and vise versa
-        try:
-            y, x =  latlon.split(",")
-        except:
-            return False, ('Not Found', 400, {'x-error': 'Both latitude and longitude must be specified to search for'
-                                                         ' a point'}), []
-        lat = float(y)
-        lon = float(x)
-        pt = [lat, lon]
-
-        # check format of coordinates, if provided
-        if lat is not None and len is not None:
-            # check latitude
-            if re.match("^(\+|-)?(?:90(?:(?:\.0{1,})?)|(?:[0-9]|[1-8][0-9])(?:(?:\.[0-9]{1,})?))$", lat) is None:
-                return False, ('Not Found', 400, {'x-error': 'Invalid latitude'}), []
-
-            # check longitude
-            if re.match("^(\+|-)?(?:180(?:(?:\.0{1,})?)|(?:[0-9]|[1-9][0-9]|1[0-7][0-9])(?:(?:\.[0-9]{1,})?))$", lon) is None:
-                return False, ('Not Found', 400, {'x-error': 'Invalid longitude'}), []
-
-        return True, '', pt
+        # lat: 0 to +/- 90, lon: 0 to +/- 180 as lat,lon
+        import re
+        latlon = kwargs.get('latLon')
+        for item in latlon.split(';'):
+            if re.match(r'^[-+]?([1-8]?\d(\.\d+)?|90(\.0+)?),\s*[-+]?(180(\.0+)?|((1[0-7]\d)|([1-9]?\d))(\.\d+)?)$',
```

Comment:
I guess this regex validate both lat and lon, but I guess we have to fix the order meaning it has to be lat first followed by separator, followed by lon, in which case doing a for loop may not be appropriate here. For example, if users give a value of "lon;lat" which will pass this regex validation even though the first lon is not a valid longitude but a valid latitude. So I think the original logic to split the latlon string into lat and lon, then validate them individually will handle this case correctly.

Reply:
Yes, the regex validates both lat 0 +/- 90, and lon 0 +/- 180 including the comma separator.
The cmaq grid covers CONUS, so I don't believe you could input a lon,lat combination that would yield a result, and you'd receive a grid coordinates of -1, -1 which would return an empty set for value or score.

Reply:
The for loop is iterating over the semicolon. More than one lat,lon pair can be submit per query, and they are separated by ;
`lat1,lon1;lat2,lon2;lat3,lon3;lat4,lon4`